einops/einops.py

```diff
@@ -624,3 +625,132 @@ def asnumpy(tensor) -> 'numpy.ndarray':
         `numpy.ndarray`, converted to numpy
     """
     return get_backend(tensor).to_numpy(tensor)
+
+
+@functools.lru_cache(256)
+def _compactify_pattern_for_einsum(pattern: str) -> str:
+    if "->" not in pattern:
```

Comment:
In several latest things I switched to using <- because output is most cases is on the left, and inputs are on the right. 

This deviates from original einops operations, but may be helpful here to make additional visible difference with native einsums.

Don't have any strong opinion on that, just curious to get your thoughts.

Reply:
I'm fine with either. Although, if I am used to using `rearrange, repeat, reduce`, or even regular `einsum`, then the first syntax I would intuitively try would be `->`. I think you wish to encourage `<-` in the future, I would allow for both `<-` and `->` patterns in all functions for the moment, and maybe give a deprecation warning in a future version that users should switch to `<-` patterns.

Just glancing at `<-`, I would assume it is an inplace operation, whereas `->` creates a new tensor. It is a bit confusing though, I feel like we should just go with `->` for everything.

@shoyer thoughts?

Reply:
One other reason my eyes seem to prefer `->` is because the same symbol is used for type declarations in function output, and any einops index string is sort of like a type declaration: the left side represents the arguments, and the right side is the output type.

Reply:
I tweaked the syntax to be `y = einsum(x, x, "i j, i k -> j k")`. Let me know if this is okay.